src/vs/workbench/browser/parts/editor/textDiffEditor.ts

```diff
@@ -60,6 +60,19 @@ export class TextDiffEditor extends BaseTextEditor implements ITextDiffEditorPan
 		this.doSaveOrClearTextDiffEditorViewState(editor);
 	}
 
+	invokeWithinContext<T>(fn: (accessor: ServicesAccessor) => T): T | null {
```

Comment:
I understand this fixes issues for @eamodio where the editor title actions are not aware of the focused side of the diff editor. But isn't this also risky to change? Here is an example:
* I install an extension that allows to open a file as binary data (hex editor)
* this extension adds to any editor and upon click opens the hex editor
* I open a diff editor from SCM
* I click on the left hand side (the right hand side is editable, left hand is not)
* I click on the Hex action
* We now treat the left hand side as active editor
* the hex action opens the binary form of the left hand editor which is just a representation of the SCM state, but not the state on disk.

In other words, this may break a lot of actions that so far have assumed the context is always the right hand diff editor. This is especially bad because we have no good indication which side of the diff editor has focus actually.

Reply:
Would the resource that it opens come from the context? Or would it be getting that some other way?

Reply:
Yeah good point, I think it actually could come in from the context we set here:

https://github.com/microsoft/vscode/blob/d50e84208f0358b4ee17b32c5394ea5605641b81/src/vs/workbench/browser/parts/editor/titleControl.ts#L232

And then is passed as static argument to the actions here:

https://github.com/microsoft/vscode/blob/d50e84208f0358b4ee17b32c5394ea5605641b81/src/vs/workbench/browser/parts/editor/titleControl.ts#L248

So it would mean that the context is always the primary (right hand side) editor but the context key service might operate on the left hand side.